src/index.jsx

```diff
@@ -7,17 +7,23 @@ import ReactDOM from 'react-dom';
 import {
   APP_INIT_ERROR, APP_READY, initialize, subscribe,
 } from '@edx/frontend-platform';
-import { ErrorPage } from '@edx/frontend-platform/react';
+import { AppProvider, ErrorPage } from '@edx/frontend-platform/react';
 import { mergeConfig } from '@edx/frontend-platform/config';
 import { messages as footerMessages } from '@edx/frontend-component-footer';
 import { messages as paragonMessages } from '@edx/paragon';
+import messages from './i18n';
 
 import { App } from './components/app';
 
 import './index.scss';
 
 subscribe(APP_READY, () => {
-  ReactDOM.render(<App />, document.getElementById('root'));
+  ReactDOM.render(
+    <AppProvider>
```

Comment:
[clarification] Why are you proposing to wrap `App` in the `AppProvider` here when `App` is the one responsible for rendering `AppProvider` today? We should only be rendering a single top-level `AppProvider` in the component tree.